### ISSUE TITLE
Unable to resize VM harddrive

### ISSUE THREAD

**Issue by ZaLiTHkA** [myspaghetti/macos-guest-virtualbox#145]
Greetings.. First off, thanks so much for putting this script together, I worked through a number of other guides on blog posts and forums, but this was the first option I actually got to work properly. :)

With that said, I'm having some difficulty trying to increase the size of my VM drive. I changed the VDI size in VirtualBox Manager to 50GB, then launched the VM to run the terminal commands as per the instructions in this repo's readme, but now I'm left with the following:

![Disk Util](https://user-images.githubusercontent.com/2149312/72200553-7b131700-3453-11ea-95cc-5974bf51bc1e.png)

My purpose for this VM is to learn my way around macOS, so I'm a little stumped as to what I need to do here. Please could someone help me with this one?

**Comment by myspaghetti** [comment]
Resizing the partition doesn't always work. Try rebooting and resizing through Disk Utility. From the menu bar select View and Show All Devices, then click on the VBOX HARDDISK Media, click the Partition button at the top center of the window, click the Partition button in the dialogue box, click on the Free Space wedge, click the minus button, and click Apply.
![delete free space partition](https://user-images.githubusercontent.com/31696646/72204383-ee328280-347f-11ea-9fb2-e2b499faf75d.jpg)

**Comment by efade** [comment]
I can confirm this issue. After resizing the virtual disk through ``VboxManage`` command, and running ``sudo diskutil repairDisk disk0``, the Free Space partition was simply not there. The new capacity even didn't get updated It still shows 26 GB even though I resized the virtual disk to 40 GB. I've rebooted the VM several times and the result is still the same.

**Comment by ZaLiTHkA** [comment]
> Resizing the partition doesn't always work...

Thanks for the screenshots, they definitely helped a bit there. I did actually have to run the `diskutil repairDisk` command a few times before it worked, didn't try that myself first in case doing so would cause other issues.

However, I've only got it partially solved now. The "VBOX HARDDISK Media" is showing the size change, but "Container disk1" and "macOS" are not.

![Disk Util - 1](https://user-images.githubusercontent.com/2149312/72205677-b848ca80-348e-11ea-84ce-88ffb53f330d.png)

![Disk Util - 2](https://user-images.githubusercontent.com/2149312/72205679-be3eab80-348e-11ea-8820-591332a104c1.png)

![Disk Util - 3](https://user-images.githubusercontent.com/2149312/72205681-c26ac900-348e-11ea-9756-4e3168a3dacc.png)

Did you do anything else to get the APFS volume size to change? After a restart, checking Disk Util again shows that the "Free Space" partition is back again, but when I try remove it now I get the following error:

![Disk Util - 4](https://user-images.githubusercontent.com/2149312/72205770-7704ea80-348f-11ea-9d6c-e71b860bb035.png)

**Comment by myspaghetti** [comment]
Yup, that happens often. Keep rebooting the guest, repairing `disk0`, and deleting the Free space partition.

**Comment by ZaLiTHkA** [comment]
So, I ended up taking what I'd learnt from this initial process and simply starting again. Fortunately I didn't remove the temporary files after running the script the _first_ time, so there wasn't much that needed to be re-downloaded.

I simply changed my drive size with `storagesize=50000` in the script file itself and everything works as expected. (: I also got a chance to shutdown my VM at a better point in the setup to export it as an "Appliance" as a fresh-OS-install backup.

---

Just a thought, but perhaps it would be worth adapting this script slightly to prompt the user for values at runtime rather than having to edit the `sh` file? I believe that would prevent quite a few issues like this (and probably many other "how do I change X" questions too).

**Comment by ZaLiTHkA** [comment]
@myspaghetti just a quick follow up from my questions earlier..

I've just finished configuring the macOS environment for hybrid mobile dev. Certain UI areas do tend to "flicker" to plain white and back though, especially anything Electron based. But I'm happy in bash, so that's just a minor inconvenience.

However, I do now have a *local* macOS environment that I can use to build hybrid apps; as I just did with a freshly created NativeScript app run inside the VM and previewed on the physical device in my hand.

Thanks for your effort with compiling this script... Really appreciate it. 👍
